File: benchmark/baselines/typescript-strict/domain.ts
export type EventState = "open" | "closed";
export type Event = Readonly<{ id: number; capacity: number; registered: number; state: EventState }>;
export type Person = Readonly<{ id: number; name: string }>;
export type Registration = Readonly<{ eventId: number; personId: number; createdAt: number }>;
export type RegistrationError = "event_full" | "already_registered" | "registration_closed" | "store_failed";
export type Result<T, E> = { readonly ok: true; readonly value: T } | { readonly ok: false; readonly error: E };

File: benchmark/baselines/rust/domain.rs
#[derive(Clone, Debug, PartialEq)] pub enum EventState { Open, Closed }
#[derive(Clone)] pub struct Event { pub id: i64, pub capacity: i64, pub registered: i64, pub state: EventState }
#[derive(Clone)] pub struct Person { pub id: i64, pub name: String }
#[derive(Clone, Debug, PartialEq)] pub struct Registration { pub event_id: i64, pub person_id: i64, pub created_at: i64 }
#[derive(Clone, Debug, PartialEq)] pub enum RegistrationError { EventFull, AlreadyRegistered, RegistrationClosed, StoreFailed }
